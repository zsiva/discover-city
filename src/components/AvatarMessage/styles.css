.avatarWrapper {
  overflow: hidden;
  margin: 1rem auto;
}
.avatarImage {
  float: left;
}
.avatarMessage {
  text-align: center;
  float: left;
  background: white;
  padding: 1rem;
  border-radius: 5px;
  width: 195px;
  margin-left: 1rem;
}

@media only screen and (min-width: 481px) {
  .avatarMessage {
    width: 240px;
  }
}
