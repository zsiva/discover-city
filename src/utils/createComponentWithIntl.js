import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import messages from '../data/messages';
import flattenMessages from '../utils/flattenMessages';

const createComponentWithIntl = (children, props = { locale: 'en-US' }) => {
  return renderer.create(
    <IntlProvider {...props} messages={flattenMessages(messages[props.locale])}>
      {children}
    </IntlProvider>,
  );
};

export default createComponentWithIntl;
