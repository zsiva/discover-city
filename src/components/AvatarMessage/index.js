//@flow
import React, { type Node, Fragment } from 'react';
import { Image } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';
import './styles.css';

export type AvatarMessagePropType = {
  imgSrc: string,
  introText: string,
  children?: Node,
};

const AvatarMessage = (props: AvatarMessagePropType) => {
  const { imgSrc, introText } = props;

  return (
    <Fragment>
      <Image circular size="tiny" src={imgSrc} className="avatarImage" />
      <div className="avatarMessage">
        <p>
          <b>
            <FormattedMessage id={introText} />
          </b>
        </p>
        {props.children}
      </div>
    </Fragment>
  );
};

export default AvatarMessage;
