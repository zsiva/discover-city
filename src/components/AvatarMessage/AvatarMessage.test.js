import React from 'react';
import renderer from 'react-test-renderer';

import createComponentWithIntl from '../../utils/createComponentWithIntl';
import AvatarMessage from './';

describe('AvatarMessage', () => {
  it('should match snapshot', () => {
    const component = createComponentWithIntl(
      <AvatarMessage imgSrc="./images/receptionist.png" introText="casino.welcome" />,
      { locale: 'en-US' },
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
