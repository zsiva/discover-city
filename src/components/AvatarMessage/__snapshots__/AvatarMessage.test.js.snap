// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarMessage should match snapshot 1`] = `
<div
  className="avatarWrapper"
>
  <img
    className="ui tiny circular image avatarImage"
    src="./images/receptionist.png"
  />
  <div
    className="avatarMessage"
  >
    <p>
      <b>
        What can I do for you?
      </b>
    </p>
  </div>
</div>
`;
